lms/djangoapps/courseware/module_render.py

```diff
@@ -491,10 +490,13 @@ def get_event_handler(event_type):
         Returns None if no special processing is required.
         """
         handlers = {
-            'completion': handle_completion_event,
             'grade': handle_grade_event,
-            'progress': handle_deprecated_progress_event,
         }
+        if completion_waffle.waffle().is_enabled(completion_waffle.ENABLE_COMPLETION_TRACKING):
+            handlers.update({
+                'completion': handle_completion_event,
+                'progress': handle_deprecated_progress_event,
+            })
```

Comment:
Changed this to better match current upstream behaviour. 

Also, the original code throws an error for blocks that emit a completion event in a handler but the completion is not enabled.